lib/Dialect/Openfhe/Transforms/ConfigureCryptoContext.cpp

```diff
@@ -0,0 +1,161 @@
+#include "lib/Dialect/Openfhe/Transforms/ConfigureCryptoContext.h"
+
+#include <cstddef>
+#include <string>
+
+#include "lib/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "lib/Dialect/Openfhe/IR/OpenfheTypes.h"
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace openfhe {
+
+#define GEN_PASS_DEF_CONFIGURECRYPTOCONTEXT
+#include "lib/Dialect/Openfhe/Transforms/Passes.h.inc"
+
+// Helper function to check if the function has MulOp or MulNoRelinOp
+bool hasMulOp(func::FuncOp op) {
+  bool result = false;
+  op.walk<WalkOrder::PreOrder>([&](Operation *op) {
+    if (isa<openfhe::MulOp>(op) || isa<openfhe::MulNoRelinOp>(op) ||
+        isa<openfhe::MulPlainOp>(op)) {
+      result = true;
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+  return result;
+}
+
+// Helper function to find all the rotation indices in the function
+SmallVector<int64_t> findAllRotIndices(func::FuncOp op) {
+  SmallVector<int64_t> rotIndices;
+  op.walk([&](openfhe::RotOp rotOp) {
+    // TODO : I'm not sure if this is the right way to get the index
+    auto typedIndex =
+        dyn_cast<arith::ConstantOp>(rotOp.getIndex().getDefiningOp())
+            .getValue();
+    int64_t rotIndex = dyn_cast<IntegerAttr>(typedIndex).getInt();
+    if (std::find(rotIndices.begin(), rotIndices.end(), rotIndex) ==
```

Comment:
I think you could probably accumulate these in a set and then convert into a smallvector if you need!